app/javascript/authentication/SignInForm.tsx

```diff
@@ -54,13 +55,57 @@ const NewAccountNotConfirmedModal = ({
           {emailSentError}
         </Alert>
       )}
+    </>
+  )
+}
+
+const ExpiredUnconfirmedModal = ({ email, onClose }: { email: string; onClose: () => void }) => {
```

Comment:
@cade-exygy  had suggested putting these in their own utility file, and I like that idea - could we do that?